distribution/lib/Standard/Base/0.0.0-dev/src/Runtime/Ref.enso

```diff
@@ -8,16 +9,27 @@ type Ref
     ## PRIVATE
        ADVANCED
        Creates a new reference containing the provided value.
+       If `allow_gc` is set to `True` than the management
+       of the reference is controlled by the runtime which
+       may _set the reference to `Nothing` any time_. The
+       definition of _"any time"_ is intentionally vague, but
+       the known implementations reset the value when running
+       out of memory or when a user asks for _"clean re-execution"_
+       of a project.
 
        Arguments:
        - value: The value to be contained in the ref.
+       - allow_gc: Is the value eligible for being cleared by the system at any time?
```

Comment:
Could we replace this with an atom type `reference_type`?

e.g.
```
type Reference_Type
    Weak
    Regular
```

Then if we ever want to also add support for soft references, it is just the matter of adding yet another variant to `Reference_Type`.

Reply:
- I will try to get this working with Greg's #11342
- then we'll see what's actually needed

Reply:
Right now I am thinking about:
```
type Ref_Auto_Clean
       Never # regular reference
       Explicit # only per user request
       Soon # using `WeakReference` disappears as soon as "possible"
       Managed estimated_memory_size:Integer time_in_ms_to_recompute:Integer # automatic when needed
```
the `..Explicit` cleaning policy would solve @GregoryTravis's 

>  the reference ... collected, then I know that the **refresh button has been clicked**

We would connect cleanup of `..Explicit` directly to **refresh button click** and no GC.

Or we could map all above types to estimates of memory and recomputation time:
- `Never` - zero memory, maximal  time
- `Explicit` - zero memory, zero time
- `Soon` - some memory, zero time
- `Managed` - some memory and some time

Reply:
Not needed anymore.